Orange/widgets/visualize/owdistributions.py

```diff
@@ -227,34 +228,25 @@ def set_data(self, data):
                 self.warning("Empty input data cannot be visualized")
                 return
             domain = self.data.domain
-            self.varmodel[:] = list(domain.variables) + \
-                               [meta for meta in domain.metas
-                                if meta.is_continuous or meta.is_discrete]
+            self.varmodel.set_domain(domain)
             self.groupvarview.clear()
-            self.groupvarmodel = \
-                ["(None)"] + [var for var in domain.variables if var.is_discrete] + \
-                [meta for meta in domain.metas if meta.is_discrete]
-            self.groupvarview.addItem("(None)")
-            for var in self.groupvarmodel[1:]:
-                self.groupvarview.addItem(self.icons[var], var.name)
+            self.groupvarmodel.set_domain(domain)
             if domain.has_discrete_class:
-                self.groupvar_idx = \
-                    self.groupvarmodel[1:].index(domain.class_var) + 1
+                self.group_var = domain.class_var
             self.openContext(domain)
-            self.variable_idx = min(max(self.variable_idx, 0),
-                                    len(self.varmodel) - 1)
-            self.groupvar_idx = min(max(self.groupvar_idx, 0),
-                                    len(self.groupvarmodel) - 1)
-            itemmodels.select_row(self.varview, self.variable_idx)
+            self.group_var = None if self.group_var not in self.groupvarmodel else self.group_var
+            self.var = self.var if self.var in self.varmodel else None
+            self.group_var = self.group_var if self.group_var in self.groupvarmodel else None
+            itemmodels.select_row(self.varview, self.var)
```

Comment:
AFAIK, the second argument for `select_row` must be an `int`. Anyway, when the data is loaded, a visualization is shown, but nothing is selected in the listview - I suspect that because of this function.